airflow/providers/common/sql/hooks/sql.py

```diff
@@ -244,7 +244,9 @@ def split_sql_string(sql: str) -> List[str]:
         :return: list of individual expressions
         """
         splits = sqlparse.split(sqlparse.format(sql, strip_comments=True))
-        statements = [s.rstrip(';') for s in splits if s.endswith(';')]
+        statements: List[str] = list(
+            filter(None, [s.rstrip(';') if s.endswith(';') else s.strip() for s in splits])
+        )
```

Comment:
```suggestion
        statements = list(
            filter(None, (s.rstrip(';') if s.endswith(';') else s for s in splits))
        )
```

The `strip()` in the `else` part isn’t consistent. Alternatively, `strip()` should be done for both cases:

```python
statements = (s.rstrip(';') if s.endswith(';') else s for s in splits)
statements = (s.strip() for s in statements)
return [s for s in statements if s]
```

Reply:
Ah yeah. Lost it in the last change :)

Reply:
Fixed and added test.

Reply:
> Sorry for being offline. I tried to work on #25259 finally.

No worries :)